node-repository/src/main/java/com/yahoo/vespa/hosted/provision/provisioning/Preparer.java

```diff
@@ -122,7 +122,8 @@ private void moveToActiveGroup(List<Node> surplusNodes, int wantedGroups, Option
     private int findHighestIndex(ApplicationId application, ClusterSpec cluster) {
         int highestIndex = -1;
         for (Node node : nodeRepository.getNodes(application,
-                                                 Node.State.active, Node.State.inactive, Node.State.parked, Node.State.failed)) {
+                                                 Node.State.reserved, Node.State.active, Node.State.inactive,
```

Comment:
This should just include all states for which `Node.State#isAllocated` is `true`.